src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -2,39 +2,42 @@
 
 namespace Microsoft.CST.OpenSource.PackageManagers
 {
-    using Helpers;
-    using HtmlAgilityPack;
+    using Contracts;
     using PackageUrl;
     using Model;
+    using Model.Metadata;
+    using Model.Providers;
     using NuGet.Packaging;
     using NuGet.Packaging.Core;
+    using NuGet.Versioning;
     using System;
     using System.Collections.Generic;
     using System.IO;
     using System.Linq;
     using System.Net.Http;
     using System.Text.Json;
     using System.Threading.Tasks;
-    using Utilities;
-    using Version = SemanticVersioning.Version;
-
 
     public class NuGetProjectManager : BaseProjectManager
     {
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0044:Add readonly modifier", Justification = "Modified through reflection.")]
         public const string ENV_NUGET_ENDPOINT_API = "https://api.nuget.org";
         public const string ENV_NUGET_ENDPOINT = "https://www.nuget.org";
         public const string NUGET_DEFAULT_REGISTRATION_ENDPOINT = "https://api.nuget.org/v3/registration5-gz-semver2/";
+        private const string NUGET_DEFAULT_CONTENT_ENDPOINT = "https://api.nuget.org/v3-flatcontainer/";
 
         private string? RegistrationEndpoint { get; set; } = null;
+        private IManagerProvider ManagerProvider { get; }
 
-        public NuGetProjectManager(IHttpClientFactory httpClientFactory, string destinationDirectory) : base(httpClientFactory, destinationDirectory)
+        public NuGetProjectManager(IHttpClientFactory httpClientFactory, string destinationDirectory, IManagerProvider? provider = null) : base(httpClientFactory, destinationDirectory)
         {
+            ManagerProvider = provider ?? new NuGetProvider();
             GetRegistrationEndpointAsync().Wait();
         }
 
         public NuGetProjectManager(string destinationDirectory) : base(destinationDirectory)
         {
+            ManagerProvider = new NuGetProvider();
             GetRegistrationEndpointAsync().Wait();
         }
```

Comment:
Nit: Could we simply make the `IManagerProvider` required? It seems like the only existing uses always supply the argument.